ocp_resources/ip_address_pool.py

```diff
@@ -0,0 +1,53 @@
+# API reference: https://metallb.universe.tf/apis/#metallb.io/v1beta1.IPAddressPool
+
+from ocp_resources.resource import NamespacedResource
+
+
+class IPAddressPool(NamespacedResource):
+    """
+    IPAddressPool object.
+    """
+
+    api_group = NamespacedResource.ApiGroup.METALLB_IO
+
+    def __init__(
+        self,
+        name,
+        addresses,
```

Comment:
all args are to be defined as "None" (if a yaml is passed, none are relevant).

Reply:
`addresses` is mandatory field, that's the reason I removed the `None`.

Reply:
See my original comment; this is not the way to check that `addresses` is mandatory)

Reply:
Replace: `if self.addresses` with `self.addresses` which will make it mandatory.

Reply:
Updated it.